resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
-                self.__read_settings()
-            else:
-                self.__write_settings()
-        else:
-            # data directory doesn't exist
+                shutil.move(Path.join(self._setting['data_directory'],
+                                     self.CONFIG_FILENAME)
+                            , Path.join(self._setting['data_directory'],
+                                     "{0}.{1}".format(self.CONFIG_FILENAME,'old')))
             self.__write_settings()
+            raise exceptions.InvalidSettings(Path.join(
+                                            self._setting['data_directory'],
+                                            self.CONFIG_FILENAME))
```

Comment:
Why do you go in all this trouble to recreate the file while keeping a backup and then raise an exception? If it's to warn the user, then warn the user, don't crash :)